account_payment_mode_brand/models/account_invoice.py

```diff
@@ -0,0 +1,25 @@
+# Copyright 2020 ACSONE SA/NV
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, models
+
+
+class AccountInvoice(models.Model):
+
+    _name = "account.invoice"
+    _inherit = ["account.invoice", "res.brand.allowed.payment.mode.mixin"]
+
+    @api.onchange("partner_id", "company_id")
+    def _onchange_partner_id(self):
```

Comment:
Why do you need this `onchange("partner_id")` ?
An `onchange("payment_mode_id")` in the mixin couldn't do the job ?

Reply:
I need to unset the payment mode when the partner is changed and the partner default payment mode is not in the brand allowed payment modes. 

It can't be done in the mixin as the partner field is defined in the sub-model (invoice model for this example.)

Reply:
You are checking `self.payment_mode_id not in self.brand_id.allowed_payment_mode_ids`.
This check should be performed when `self.payment_mode_id` or `self.brand_id.allowed_payment_mode_ids` is changed.
Thus the onchange should be on `payment_mode_id` and `brand_id`, right ?

My point is if changing `partner_id` induces a change on `payment_mode_id`, `onchange("payment_mode_id")` should catch it.
If changing `partner_id` doesn't induces a change on `payment_mode_id`, we don't need this method to be called anyway.

Reply:
I see your point now, I will make the changes